src/api/pastel-rpc/sent-tx-store.ts

```diff
@@ -0,0 +1,109 @@
+import { remote } from 'electron'
+import fs from 'fs'
+import os from 'os'
+import path from 'path'
+
+import {
+  TDetailedTxns,
+  TListTransactions,
+  TSentTxStore,
+  TVjoinsplit,
+} from './network-stats/type'
+
+const locateSentTxStore = (): string => {
+  if (os.platform() === 'darwin') {
+    return path.join(remote.app.getPath('appData'), 'Pastel', 'senttxstore.dat')
+  }
+
+  if (os.platform() === 'linux') {
+    return path.join(
+      remote.app.getPath('home'),
+      '.local',
+      'share',
+      'psl-qt-wallet-org',
+      'psl-qt-wallet',
+      'senttxstore.dat',
+    )
+  }
+
+  return path.join(remote.app.getPath('appData'), 'Pastel', 'senttxstore.dat')
+}
+
+export const loadSentTxns = async (): Promise<TListTransactions | []> => {
+  try {
+    const sentTx = JSON.parse(
+      (await fs.promises.readFile(locateSentTxStore())).toString(),
+    )
+    return sentTx.map((s: TSentTxStore) => {
+      const transction: TListTransactions = new Transaction()
+      transction.type = s.type
+      transction.amount = s.amount
+      transction.address = s.from
+      transction.txid = s.txid
+      transction.time = s.datetime
+      transction.detailedTxns = [new TxDetail()]
+      transction.detailedTxns[0].address = s.address
+      transction.detailedTxns[0].amount = s.amount
+      transction.detailedTxns[0].memo = s.memo
+      return transction
+    })
+  } catch (err) {
+    // If error for whatever reason (most likely, file not found), just return an empty array
+    return []
+  }
+}
```

Comment:
This is not related to RPC. Can we move it to the feature that is using this function?